
<!doctype html>
<html lang="en" class="no-js">
  <head>
    
      <meta charset="utf-8">
      <meta name="viewport" content="width=device-width,initial-scale=1">
      
      
      
      <link rel="icon" href="../../../assets/images/favicon.png">
      <meta name="generator" content="mkdocs-1.3.1, mkdocs-material-8.2.11">
    
    
      
        <title>Variant calling gatk1 print - Bioinformatics Documentation</title>
      
    
    
      <link rel="stylesheet" href="../../../assets/stylesheets/main.8c5ef100.min.css">
      
        
        <link rel="stylesheet" href="../../../assets/stylesheets/palette.9647289d.min.css">
        
      
    
    
    
      
        
        
        <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin>
        <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Roboto:300,300i,400,400i,700,700i%7CRoboto+Mono:400,400i,700,700i&display=fallback">
        <style>:root{--md-text-font:"Roboto";--md-code-font:"Roboto Mono"}</style>
      
    
    
      <link rel="stylesheet" href="../../../extra.css">
    
    <script>__md_scope=new URL("../../..",location),__md_get=(e,_=localStorage,t=__md_scope)=>JSON.parse(_.getItem(t.pathname+"."+e)),__md_set=(e,_,t=localStorage,a=__md_scope)=>{try{t.setItem(a.pathname+"."+e,JSON.stringify(_))}catch(e){}}</script>
    
<script async src="https://www.googletagmanager.com/gtag/js?id=G-EMNLJD0DWP"></script>
<script>
    window.dataLayer = window.dataLayer || [];
    function gtag() {
        dataLayer.push(arguments);
    }
    gtag('js', new Date());
    gtag('config', 'G-EMNLJD0DWP');
</script>

    
  </head>
  
  
    
    
    
    
    
    <body dir="ltr" data-md-color-scheme="" data-md-color-primary="none" data-md-color-accent="none">
  
    
    
    <input class="md-toggle" data-md-toggle="drawer" type="checkbox" id="__drawer" autocomplete="off">
    <input class="md-toggle" data-md-toggle="search" type="checkbox" id="__search" autocomplete="off">
    <label class="md-overlay" for="__drawer"></label>
    <div data-md-component="skip">
      
        
        <a href="#variant-calling-using-gatk4" class="md-skip">
          Skip to content
        </a>
      
    </div>
    <div data-md-component="announce">
      
    </div>
    
    
      

<header class="md-header" data-md-component="header">
  <nav class="md-header__inner md-grid" aria-label="Header">
    <a href="../../.." title="Bioinformatics Documentation" class="md-header__button md-logo" aria-label="Bioinformatics Documentation" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    <label class="md-header__button md-icon" for="__drawer">
      <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M3 6h18v2H3V6m0 5h18v2H3v-2m0 5h18v2H3v-2Z"/></svg>
    </label>
    <div class="md-header__title" data-md-component="header-title">
      <div class="md-header__ellipsis">
        <div class="md-header__topic">
          <span class="md-ellipsis">
            Bioinformatics Documentation
          </span>
        </div>
        <div class="md-header__topic" data-md-component="header-topic">
          <span class="md-ellipsis">
            
              Variant calling gatk1 print
            
          </span>
        </div>
      </div>
    </div>
    
    
    
      <label class="md-header__button md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
      </label>
      <div class="md-search" data-md-component="search" role="dialog">
  <label class="md-search__overlay" for="__search"></label>
  <div class="md-search__inner" role="search">
    <form class="md-search__form" name="search">
      <input type="text" class="md-search__input" name="query" aria-label="Search" placeholder="Search" autocapitalize="off" autocorrect="off" autocomplete="off" spellcheck="false" data-md-component="search-query" required>
      <label class="md-search__icon md-icon" for="__search">
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M9.5 3A6.5 6.5 0 0 1 16 9.5c0 1.61-.59 3.09-1.56 4.23l.27.27h.79l5 5-1.5 1.5-5-5v-.79l-.27-.27A6.516 6.516 0 0 1 9.5 16 6.5 6.5 0 0 1 3 9.5 6.5 6.5 0 0 1 9.5 3m0 2C7 5 5 7 5 9.5S7 14 9.5 14 14 12 14 9.5 12 5 9.5 5Z"/></svg>
        <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20 11v2H8l5.5 5.5-1.42 1.42L4.16 12l7.92-7.92L13.5 5.5 8 11h12Z"/></svg>
      </label>
      <nav class="md-search__options" aria-label="Search">
        
        <button type="reset" class="md-search__icon md-icon" aria-label="Clear" tabindex="-1">
          <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M19 6.41 17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12 19 6.41Z"/></svg>
        </button>
      </nav>
      
    </form>
    <div class="md-search__output">
      <div class="md-search__scrollwrap" data-md-scrollfix>
        <div class="md-search-result" data-md-component="search-result">
          <div class="md-search-result__meta">
            Initializing search
          </div>
          <ol class="md-search-result__list"></ol>
        </div>
      </div>
    </div>
  </div>
</div>
    
    
      <div class="md-header__source">
        <a href="https://github.com/melbournebioinformatics/MelBioInf_docs" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    melbournebioinformatics/MelBioInf_docs
  </div>
</a>
      </div>
    
  </nav>
  
</header>
    
    <div class="md-container" data-md-component="container">
      
      
        
          
        
      
      <main class="md-main" data-md-component="main">
        <div class="md-main__inner md-grid">
          
            
              
              <div class="md-sidebar md-sidebar--primary" data-md-component="sidebar" data-md-type="navigation" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    


<nav class="md-nav md-nav--primary" aria-label="Navigation" data-md-level="0">
  <label class="md-nav__title" for="__drawer">
    <a href="../../.." title="Bioinformatics Documentation" class="md-nav__button md-logo" aria-label="Bioinformatics Documentation" data-md-component="logo">
      
  
  <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M12 8a3 3 0 0 0 3-3 3 3 0 0 0-3-3 3 3 0 0 0-3 3 3 3 0 0 0 3 3m0 3.54C9.64 9.35 6.5 8 3 8v11c3.5 0 6.64 1.35 9 3.54 2.36-2.19 5.5-3.54 9-3.54V8c-3.5 0-6.64 1.35-9 3.54Z"/></svg>

    </a>
    Bioinformatics Documentation
  </label>
  
    <div class="md-nav__source">
      <a href="https://github.com/melbournebioinformatics/MelBioInf_docs" title="Go to repository" class="md-source" data-md-component="source">
  <div class="md-source__icon md-icon">
    
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 448 512"><!--! Font Awesome Free 6.1.1 by @fontawesome - https://fontawesome.com License - https://fontawesome.com/license/free (Icons: CC BY 4.0, Fonts: SIL OFL 1.1, Code: MIT License) Copyright 2022 Fonticons, Inc.--><path d="M439.55 236.05 244 40.45a28.87 28.87 0 0 0-40.81 0l-40.66 40.63 51.52 51.52c27.06-9.14 52.68 16.77 43.39 43.68l49.66 49.66c34.23-11.8 61.18 31 35.47 56.69-26.49 26.49-70.21-2.87-56-37.34L240.22 199v121.85c25.3 12.54 22.26 41.85 9.08 55a34.34 34.34 0 0 1-48.55 0c-17.57-17.6-11.07-46.91 11.25-56v-123c-20.8-8.51-24.6-30.74-18.64-45L142.57 101 8.45 235.14a28.86 28.86 0 0 0 0 40.81l195.61 195.6a28.86 28.86 0 0 0 40.8 0l194.69-194.69a28.86 28.86 0 0 0 0-40.81z"/></svg>
  </div>
  <div class="md-source__repository">
    melbournebioinformatics/MelBioInf_docs
  </div>
</a>
    </div>
  
  <ul class="md-nav__list" data-md-scrollfix>
    
      
      
      

  
  
  
    <li class="md-nav__item">
      <a href="../../.." class="md-nav__link">
        Home
      </a>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_2" type="checkbox" id="__nav_2" >
      
      
      
      
        <label class="md-nav__link" for="__nav_2">
          Tools and skill development
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Tools and skill development" data-md-level="1">
        <label class="md-nav__title" for="__nav_2">
          <span class="md-nav__icon md-icon"></span>
          Tools and skill development
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/bioinfo/" class="md-nav__link">
        What is Bioinformatics?
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../../guides/galaxy/" class="md-nav__link">
        Introduction to the Galaxy Platform
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../galaxy-workflows/galaxy-workflows/" class="md-nav__link">
        Introduction to Galaxy Workflows
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../python_overview/python_overview/" class="md-nav__link">
        Introduction to Python
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../using_git/Using_Git/" class="md-nav__link">
        Introduction to Git and Github
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../unix/unix/" class="md-nav__link">
        Introduction to Unix
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../intro_R_biologists/intro_R_biologists/" class="md-nav__link">
        Introduction to R for Biologists
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../Genome_browsers/GenomeBrowsers_Intro/" class="md-nav__link">
        Introduction to Genome Browsers
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cwl_2022/cwl_2022/" class="md-nav__link">
        Introduction to CWL
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../rna_seq_exp_design/rna_seq_experimental_design/" class="md-nav__link">
        RNA-seq Experimental Design
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../docker/docker/" class="md-nav__link">
        Containerized Bioinformatics
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../bionitio/bionitio/" class="md-nav__link">
        Command-line software development
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../janis_translate/janis_translate/" class="md-nav__link">
        Workflow Translation With Janis
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3" type="checkbox" id="__nav_3" >
      
      
      
      
        <label class="md-nav__link" for="__nav_3">
          Genomics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Genomics" data-md-level="1">
        <label class="md-nav__title" for="__nav_3">
          <span class="md-nav__icon md-icon"></span>
          Genomics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_1" type="checkbox" id="__nav_3_1" >
      
      
      
      
        <label class="md-nav__link" for="__nav_3_1">
          Variant detection
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Variant detection" data-md-level="2">
        <label class="md-nav__title" for="__nav_3_1">
          <span class="md-nav__icon md-icon"></span>
          Variant detection
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../var_detect_advanced/var_detect_advanced_background/" class="md-nav__link">
        Introduction to Variant detection
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../variant_calling_galaxy_1/variant_calling_galaxy_1/" class="md-nav__link">
        Variant Calling part 1 (Galaxy)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../var_detect_advanced/var_detect_advanced/" class="md-nav__link">
        Variant Calling part 2 (Galaxy)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../longread_sv_calling/longread_sv_calling/" class="md-nav__link">
        Long-read Structural Variant Calling
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../variant_calling_gatk1/" class="md-nav__link">
        Variant calling using command-line tools
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_2" type="checkbox" id="__nav_3_2" >
      
      
      
      
        <label class="md-nav__link" for="__nav_3_2">
          de novo assembly
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="de novo assembly" data-md-level="2">
        <label class="md-nav__title" for="__nav_3_2">
          <span class="md-nav__icon md-icon"></span>
          de novo assembly
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../assembly/assembly-background/" class="md-nav__link">
        Introduction to de novo assembly with Velvet
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../assembly/assembly-protocol/" class="md-nav__link">
        Introduction to de novo genome assembly for Illumina reads
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../assembly/assembly/" class="md-nav__link">
        de novo assembly of Illumina reads using Velvet (Galaxy)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../assembly/spades/" class="md-nav__link">
        de novo assembly of Illumina reads using Spades (Galaxy)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../pacbio/" class="md-nav__link">
        de novo assembly of PacBio reads (unix)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../hybrid_assembly/nanopore_assembly/" class="md-nav__link">
        Hybrid genome assembly - Nanopore and Illumina
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_3_3" type="checkbox" id="__nav_3_3" >
      
      
      
      
        <label class="md-nav__link" for="__nav_3_3">
          Metabarcoding
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Metabarcoding" data-md-level="2">
        <label class="md-nav__title" for="__nav_3_3">
          <span class="md-nav__icon md-icon"></span>
          Metabarcoding
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../qiime2/qiime2/" class="md-nav__link">
        Introduction to Metabarcoding using Qiime2
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4" type="checkbox" id="__nav_4" >
      
      
      
      
        <label class="md-nav__link" for="__nav_4">
          Transcriptomics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Transcriptomics" data-md-level="1">
        <label class="md-nav__title" for="__nav_4">
          <span class="md-nav__icon md-icon"></span>
          Transcriptomics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../rna_seq_exp_design/rna_seq_experimental_design/" class="md-nav__link">
        RNAseq Experimental Design
      </a>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_2" type="checkbox" id="__nav_4_2" >
      
      
      
      
        <label class="md-nav__link" for="__nav_4_2">
          Bulk RNAseq
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Bulk RNAseq" data-md-level="2">
        <label class="md-nav__title" for="__nav_4_2">
          <span class="md-nav__icon md-icon"></span>
          Bulk RNAseq
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../rna_seq_dge_basic/rna_seq_basic_background/" class="md-nav__link">
        Introduction to bulk RNAseq analysis
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../rna_seq_dge_basic/rna_seq_basic_tutorial/" class="md-nav__link">
        RNAseq analysis using HISAT2 (Galaxy)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../rna_seq_dge_basic/rna_seq_basic_tuxedo/" class="md-nav__link">
        RNAseq analysis using Tuxedo (Galaxy)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../rna_seq_dge_advanced/rna_seq_advanced_tutorial/" class="md-nav__link">
        RNAseq differential expression tool comparision (Galaxy)
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../rna_seq_dge_in_r/rna_seq_r/" class="md-nav__link">
        RNAseq differential expression in R
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
            
              
  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_4_3" type="checkbox" id="__nav_4_3" >
      
      
      
      
        <label class="md-nav__link" for="__nav_4_3">
          Singlecell RNAseq
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Singlecell RNAseq" data-md-level="2">
        <label class="md-nav__title" for="__nav_4_3">
          <span class="md-nav__icon md-icon"></span>
          Singlecell RNAseq
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../singlecell/" class="md-nav__link">
        10X single-cell RNA-seq analysis in R
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_5" type="checkbox" id="__nav_5" >
      
      
      
      
        <label class="md-nav__link" for="__nav_5">
          Proteomics
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Proteomics" data-md-level="1">
        <label class="md-nav__title" for="__nav_5">
          <span class="md-nav__icon md-icon"></span>
          Proteomics
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../proteomics_basic/proteomics_basic/" class="md-nav__link">
        Identifying proteins from mass spectrometry data
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_6" type="checkbox" id="__nav_6" >
      
      
      
      
        <label class="md-nav__link" for="__nav_6">
          Statistics and Visualisation
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Statistics and Visualisation" data-md-level="1">
        <label class="md-nav__title" for="__nav_6">
          <span class="md-nav__icon md-icon"></span>
          Statistics and Visualisation
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../python_viz/python_viz/" class="md-nav__link">
        Data visualisation with Python
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_7" type="checkbox" id="__nav_7" >
      
      
      
      
        <label class="md-nav__link" for="__nav_7">
          Structural Modelling
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Structural Modelling" data-md-level="1">
        <label class="md-nav__title" for="__nav_7">
          <span class="md-nav__icon md-icon"></span>
          Structural Modelling
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../molecular_dynamics_101/molecular_dynamics_101/" class="md-nav__link">
        Molecular Dynamics - Introduction to cluster computing
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../molecular_dynamics_201/molecular_dynamics_201/" class="md-nav__link">
        Molecular Dynamics - Building input files, visualising the trajectory
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_8" type="checkbox" id="__nav_8" >
      
      
      
      
        <label class="md-nav__link" for="__nav_8">
          Workshop Delivery Mode Information
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Workshop Delivery Mode Information" data-md-level="1">
        <label class="md-nav__title" for="__nav_8">
          <span class="md-nav__icon md-icon"></span>
          Workshop Delivery Mode Information
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../workshop_delivery_mode_info/workshops_nectar/" class="md-nav__link">
        Workshops on Nectar Instances
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
      
      
      

  
  
  
    
    <li class="md-nav__item md-nav__item--nested">
      
      
        <input class="md-nav__toggle md-toggle" data-md-toggle="__nav_9" type="checkbox" id="__nav_9" >
      
      
      
      
        <label class="md-nav__link" for="__nav_9">
          Archive
          <span class="md-nav__icon md-icon"></span>
        </label>
      
      <nav class="md-nav" aria-label="Archive" data-md-level="1">
        <label class="md-nav__title" for="__nav_9">
          <span class="md-nav__icon md-icon"></span>
          Archive
        </label>
        <ul class="md-nav__list" data-md-scrollfix>
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../unix_archive_2022/unix/" class="md-nav__link">
        Introduction to Unix-Archive
      </a>
    </li>
  

            
          
            
              
  
  
  
    <li class="md-nav__item">
      <a href="../../cwl/cwl/" class="md-nav__link">
        Introduction to CWL
      </a>
    </li>
  

            
          
        </ul>
      </nav>
    </li>
  

    
  </ul>
</nav>
                  </div>
                </div>
              </div>
            
            
              
              <div class="md-sidebar md-sidebar--secondary" data-md-component="sidebar" data-md-type="toc" >
                <div class="md-sidebar__scrollwrap">
                  <div class="md-sidebar__inner">
                    

<nav class="md-nav md-nav--secondary" aria-label="Table of contents">
  
  
  
    
  
  
    <label class="md-nav__title" for="__toc">
      <span class="md-nav__icon md-icon"></span>
      Table of contents
    </label>
    <ul class="md-nav__list" data-md-component="toc" data-md-scrollfix>
      
        <li class="md-nav__item">
  <a href="#author-information" class="md-nav__link">
    Author Information
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#overview" class="md-nav__link">
    Overview
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#learning-objectives" class="md-nav__link">
    Learning Objectives
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#description" class="md-nav__link">
    Description
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#requirements-and-preparation" class="md-nav__link">
    Requirements and preparation
  </a>
  
    <nav class="md-nav" aria-label="Requirements and preparation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#mode-of-delivery" class="md-nav__link">
    Mode of Delivery
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#byobu-screen" class="md-nav__link">
    Byobu-screen
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tutorial-setting" class="md-nav__link">
    Tutorial setting
  </a>
  
    <nav class="md-nav" aria-label="Tutorial setting">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#the-genome-analysis-toolkit-gatk" class="md-nav__link">
    The Genome Analysis Toolkit (GATK)
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#how-this-tutorial-works" class="md-nav__link">
    How this tutorial works
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#tutorial-contents-table" class="md-nav__link">
    Tutorial contents table
  </a>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-1-map-raw-mapped-reads-to-reference-genome" class="md-nav__link">
    Section 1: Map raw mapped reads to reference genome
  </a>
  
    <nav class="md-nav" aria-label="Section 1: Map raw mapped reads to reference genome">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-preparation-and-data-import" class="md-nav__link">
    1. Preparation and data import
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-align-genome" class="md-nav__link">
    2. Align genome
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-2-prepare-analysis-ready-reads" class="md-nav__link">
    Section 2: Prepare analysis ready reads
  </a>
  
    <nav class="md-nav" aria-label="Section 2: Prepare analysis ready reads">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-sort-sambam" class="md-nav__link">
    1. Sort SAM/BAM
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-mark-duplicate-reads" class="md-nav__link">
    2. Mark duplicate reads
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-base-quality-recalibration" class="md-nav__link">
    3. Base quality recalibration
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-3-variant-calling" class="md-nav__link">
    Section 3: Variant calling
  </a>
  
    <nav class="md-nav" aria-label="Section 3: Variant calling">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-apply-haplotypecaller" class="md-nav__link">
    1. Apply HaplotypeCaller
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-apply-combinegvcfs" class="md-nav__link">
    2. Apply CombineGVCFs
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-apply-genotypegvcfs" class="md-nav__link">
    3. Apply GenotypeGVCFs
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-4-filter-and-prepare-analysis-ready-variants" class="md-nav__link">
    Section 4: Filter and prepare analysis ready variants
  </a>
  
    <nav class="md-nav" aria-label="Section 4: Filter and prepare analysis ready variants">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-variant-quality-score-recalibration" class="md-nav__link">
    1. Variant Quality Score Recalibration
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-additional-filtering" class="md-nav__link">
    2. Additional filtering
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#3-final-analysis-ready-vcf-file" class="md-nav__link">
    3. Final analysis ready VCF file
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
        <li class="md-nav__item">
  <a href="#section-5-exporting-variant-data-and-visualisation" class="md-nav__link">
    Section 5: Exporting variant data and visualisation
  </a>
  
    <nav class="md-nav" aria-label="Section 5: Exporting variant data and visualisation">
      <ul class="md-nav__list">
        
          <li class="md-nav__item">
  <a href="#1-variantstotable" class="md-nav__link">
    1. VariantsToTable
  </a>
  
</li>
        
          <li class="md-nav__item">
  <a href="#2-html-report" class="md-nav__link">
    2. HTML report
  </a>
  
</li>
        
      </ul>
    </nav>
  
</li>
      
    </ul>
  
</nav>
                  </div>
                </div>
              </div>
            
          
          <div class="md-content" data-md-component="content">
            <article class="md-content__inner md-typeset">
              
                
  <a href="https://github.com/melbournebioinformatics/MelBioInf_docs/edit/master/docs/tutorials/variant_calling_gatk1/variant_calling_gatk1_print.md" title="Edit this page" class="md-content__button md-icon">
    <svg xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><path d="M20.71 7.04c.39-.39.39-1.04 0-1.41l-2.34-2.34c-.37-.39-1.02-.39-1.41 0l-1.84 1.83 3.75 3.75M3 17.25V21h3.75L17.81 9.93l-3.75-3.75L3 17.25Z"/></svg>
  </a>



<p><img alt="melbioinf_logo" src="../../media/melbioinf_logo.png" style="width:350px; padding-right:50px" />       <img alt="unimelb_logo" src="../../media/PRIMARY_A_Vertical_Housed_RGB.png" style="width:150px" /></p>
<!-- pre code {
  white-space: pre-wrap;
} -->

<h1 id="variant-calling-using-gatk4">Variant calling using GATK4<a class="headerlink" href="#variant-calling-using-gatk4" title="Permanent link">&para;</a></h1>
<!-- <mark>This workshop is under development and testing and may contain bugs.</mark> -->
<!-- {​==**This workshop is under development and testing and may contain bugs.**==} -->

<p>Anticipated workshop duration when delivered to a group of participants is <strong>4 hours</strong>.</p>
<p>For queries relating to this workshop, contact Melbourne Bioinformatics (<a href="&#109;&#97;&#105;&#108;&#116;&#111;&#58;&#98;&#105;&#111;&#105;&#110;&#102;&#111;&#114;&#109;&#97;&#116;&#105;&#99;&#115;&#45;&#116;&#114;&#97;&#105;&#110;&#105;&#110;&#103;&#64;&#117;&#110;&#105;&#109;&#101;&#108;&#98;&#46;&#101;&#100;&#117;&#46;&#97;&#117;">&#98;&#105;&#111;&#105;&#110;&#102;&#111;&#114;&#109;&#97;&#116;&#105;&#99;&#115;&#45;&#116;&#114;&#97;&#105;&#110;&#105;&#110;&#103;&#64;&#117;&#110;&#105;&#109;&#101;&#108;&#98;&#46;&#101;&#100;&#117;&#46;&#97;&#117;</a>).</p>
<h2 id="author-information">Author Information<a class="headerlink" href="#author-information" title="Permanent link">&para;</a></h2>
<p>Khalid Mahmood<br />
Melbourne Bioinformatics, University of Melbourne<br />
Developed: September 2021<br />
Reviewed: June 2023</p>
<h2 id="overview">Overview<a class="headerlink" href="#overview" title="Permanent link">&para;</a></h2>
<p><em>Topic</em></p>
<ul class="task-list">
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Genomics</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled/><span class="task-list-indicator"></span></label> Transcriptomics</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled/><span class="task-list-indicator"></span></label> Proteomics</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled/><span class="task-list-indicator"></span></label> Metabolomics</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled/><span class="task-list-indicator"></span></label> Statistics and visualisation</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled/><span class="task-list-indicator"></span></label> Structural Modelling</li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled/><span class="task-list-indicator"></span></label> Basic skills</li>
</ul>
<p><em>Skill level</em></p>
<ul class="task-list">
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled/><span class="task-list-indicator"></span></label> Beginner  </li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled checked/><span class="task-list-indicator"></span></label> Intermediate  </li>
<li class="task-list-item"><label class="task-list-control"><input type="checkbox" disabled/><span class="task-list-indicator"></span></label> Advanced  </li>
</ul>
<p><br>
This workshop is designed for participants with some command-line knowledge. You will need to be able to <code>ssh</code> into a remote machine, navigate the directory structure and <code>scp</code> files from a remote computer to your local computer.</p>
<h2 id="learning-objectives">Learning Objectives<a class="headerlink" href="#learning-objectives" title="Permanent link">&para;</a></h2>
<p>At the end of this workshop, you will be able to:</p>
<ul>
<li>Take raw DNA sequencing reads and perform variant calling to produce a variant list using GATK4.</li>
<li>Perform basic exploration of variants.</li>
</ul>
<h2 id="description">Description<a class="headerlink" href="#description" title="Permanent link">&para;</a></h2>
<p>This tutorial runs through the GATK4 best practices workflow for variant calling. The workflow starts and a pair of sequencing reads and performs a series of steps to determine a set to genetic variants.</p>
<p><strong>Data:</strong> Illumina HiSeq paired-end (2×100 bp) reads in FASTQ format.<br />
<strong>Tools:</strong> GATK4, Picard, Bcftools and jigv<br />
<strong>Reference data:</strong> GATK4 hg38 reference bundle and hg38 refGene annotation (hg38.refGene.gtf.gz)</p>
<h2 id="requirements-and-preparation">Requirements and preparation<a class="headerlink" href="#requirements-and-preparation" title="Permanent link">&para;</a></h2>
<div class="admonition attention">
<p class="admonition-title">Important</p>
<p><strong>Attendees are required to use their own laptop computers.</strong>  </p>
<p>At least one week before the workshop, if required, participants should install the software below.  This should provide sufficient time for participants to liaise with their own IT support should they encounter any IT problems.  </p>
</div>
<p><strong>Software</strong>  </p>
<p><em>Mac Users:</em> No additional software needs to be installed for this workshop.</p>
<p><em>Windows Users:</em><br />
1. A terminal emulator such as <a href="https://www.chiark.greenend.org.uk/~sgtatham/putty/latest.html">PuTTY</a>(free and open-source) will need to be downloaded.<br />
2. Software for file transfers between a local computer and remote server such as <a href="https://winscp.net/eng/index.php">WinSCP</a> or <a href="https://filezilla-project.org/">FileZilla</a>.</p>
<p><strong>Data</strong></p>
<ul>
<li>All required data will be made available on the workshop virtual machines (hosted at the Melbourne Research Cloud). Login details will be provided closer to the tutorial data.</li>
</ul>
<!-- * If you wish to analyse the data independently at a later stage, it can be downloaded from [here](https://github.com/melbournebioinformatics/MelBioInf_docs/blob/29266406cb16cf376ef5f3d48e9bf8ac3578f1b0/docs/tutorials/qiime2/raw_data.zip). This zipped folder contains both the FASTQs and associated metadata file.     -->

<h3 id="mode-of-delivery">Mode of Delivery<a class="headerlink" href="#mode-of-delivery" title="Permanent link">&para;</a></h3>
<p>This workshop will be run on a <a href="https://cloud.nectar.org.au/">Nectar</a> Instance. An “Instance” is Nectar terminology for a virtual machine running on the Nectar Cloud OpenStack infrastructure. An “Instance” runs on a “compute node”; i.e. a physical computer populated with processor chips, memory chips and so on.</p>
<p>You will be given an individual username, IP address and password to log on to using the SSH client tool on your computer (Terminal on Mac or PuTTY on Windows).</p>
<div class="highlight"><pre><span></span><code>ssh username@ip-address
</code></pre></div>
<!-- <br>
Should you wish to do this tutorial at a later stage independently, it is possible to apply for your own instance directly through a [Nectar allocation](https://support.ehelp.edu.au/support/solutions/articles/6000068044-managing-an-allocation). There are also many helpful [Nectar Research Cloud tutorials](https://tutorials.rc.nectar.org.au/). -->

<h3 id="byobu-screen">Byobu-screen<a class="headerlink" href="#byobu-screen" title="Permanent link">&para;</a></h3>
<p>Some of the commands in this tutorial take a while to run. Should your ssh connection drop and the SSH session on Nectar terminates, any commands that are running will terminate too. To mitigate this, once logged on to the Nectar Instance, we&rsquo;ll run <code>byobu-screen</code> (an enhancement for the <code>screen</code> terminal multiplexer) which allows us to resume a session. In other words, processes running in <code>byobu-screen</code> will continue to run when their window is not visible, even if you get disconnected.</p>
<p>On Nectar, to start a <code>byobu-screen</code> session called <code>workshop</code>, type  </p>
<div class="highlight"><pre><span></span><code>byobu-screen -S workshop
</code></pre></div>
<p>You can then proceed to run the commands in the workshop as normal.</p>
<details class="example">
<summary>Help: Byobu basics</summary>
<p>The screen should look like
<img alt="Byobu" src="../media/byobu.png" />
<br>
Should your SSH session on Nectar terminate, once you log back in to your Nectar instance, list running sessions/screens:</p>
<div class="highlight"><pre><span></span><code>byobu-screen -ls
</code></pre></div>
<p>If it says (Detached) next to the <code>workshop</code> session in the list, reattach to <code>workshop</code> by:</p>
<div class="highlight"><pre><span></span><code>byobu-screen -r workshop
</code></pre></div>
<p>If it says (Attached) next to the <code>workshop</code> session in the list, you can access the <code>workshop</code> session which is already attached by:</p>
<div class="highlight"><pre><span></span><code>byobu-screen -r -d workshop
</code></pre></div>
<p><br>
Some other useful <code>byobu-screen</code> commands:</p>
<ul>
<li>
<p>To detach from <code>workshop</code>, type <code>ctrl-a ctrl-d</code> while inside the <code>workshop</code> session.
(You will need to configure Byobu&rsquo;s ctrl-a behaviour if it hasn&rsquo;t already been configured (text will appear on the screen telling you this). Follow the information on the screen and select <code>1</code> for Screen mode).</p>
</li>
<li>
<p>To terminate <code>workshop</code>, type <code>ctrl-d</code> while inside the <code>workshop</code> session.</p>
</li>
</ul>
</details>
<h2 id="tutorial-setting">Tutorial setting<a class="headerlink" href="#tutorial-setting" title="Permanent link">&para;</a></h2>
<p>The setting for this tutorial is as follows: you receive some germline sequencing data from an individual <a href="https://www.internationalgenome.org/data-portal/sample/NA12878">NA12878</a> (chr20 from sample NA12878 is sourced from the International Genome Sample Resource). Your aim is to identify small genetic variants from this data.</p>
<h3 id="the-genome-analysis-toolkit-gatk">The Genome Analysis Toolkit (GATK)<a class="headerlink" href="#the-genome-analysis-toolkit-gatk" title="Permanent link">&para;</a></h3>
<p>GATK is the industry standard toolkit for analysis of germline DNA to identify SNVs and indels. The GATK tool is mainly designed fo the human whole genome and exome analysis. The current version (GATK4) has expanded scope now and includes more complex analysis such copy number (CNV), structural variant (SV) and somatic variants. The current version of GATK also includes several utility functions for processing alignment files, VCF files and other complex processing workflows. The Picard toolkit is also now incorporated with GATK4.
Van der Auwera GA &amp; O&rsquo;Connor BD. (2020). Genomics in the Cloud: Using Docker, GATK, and WDL in Terra (1<sup>st</sup> Edition). O&rsquo;Reilly Media.</p>
<h3 id="how-this-tutorial-works">How this tutorial works<a class="headerlink" href="#how-this-tutorial-works" title="Permanent link">&para;</a></h3>
<div class="highlight"><pre><span></span><code>These grey coloured boxes are code blocks. The rectangular boxes in the top
right hand corner of this code block/grey box can be used to copy the code to
the clipboard.
</code></pre></div>
<details class="example">
<summary><strong>Click here</strong> </summary>
<p>REVEALED!
Coloured boxes like these when clicked will reveal their content.</p>
</details>
<p>To copy data from the remote workshop computer to your local computer.</p>
<p><strong>Command line/Mac Users</strong></p>
<p>The syntax to do this depends on whether you are running the copying command on your local computer, or on the remote computer (Nectar cloud).</p>
<ol>
<li>
<p>When running the command from your local computer, the syntax for copying a file <em>from</em> Nectar is:</p>
<div class="highlight"><pre><span></span><code>scp username@nectar_IP_address:FILENAME /PATH/TO/TARGET/FOLDER/
</code></pre></div>
</li>
<li>
<p>Running the command on the remote computer, the syntax for copying a file <em>to</em> your local computer is:
    <div class="highlight"><pre><span></span><code>scp FILENAME username@your_IP_address:/PATH/TO/TARGET/FOLDER/
</code></pre></div></p>
</li>
</ol>
<p><strong>SFTP clients</strong></p>
<p>Using WinSCP or FileZilla or CyberDuck you will need the following details to connect and browse.</p>
<ol>
<li>Host: The IP address of the Nectar instance</li>
<li>Username:</li>
<li>Port:</li>
</ol>
<h2 id="tutorial-contents-table">Tutorial contents table<a class="headerlink" href="#tutorial-contents-table" title="Permanent link">&para;</a></h2>
<ul>
<li><a href="#section-1-map-raw-mapped-reads-to-reference-genome">Section 1: Map raw mapped reads to reference genome</a></li>
<li><a href="#section-2-prepare-analysis-ready-reads">Section 2: Prepare analysis ready reads</a></li>
<li><a href="#section-3-variant-calling">Section 3: Variant calling</a></li>
<li><a href="#section-4-filter-and-prepare-analysis-ready-variants">Section 4: Filter and prepare analysis ready variants</a></li>
<li><a href="#section-5-exporting-variant-data-and-visualisation">Section 5: Exporting variant data and visualisation</a></li>
</ul>
<h2 id="section-1-map-raw-mapped-reads-to-reference-genome">Section 1: Map raw mapped reads to reference genome<a class="headerlink" href="#section-1-map-raw-mapped-reads-to-reference-genome" title="Permanent link">&para;</a></h2>
<h3 id="1-preparation-and-data-import">1. Preparation and data import<a class="headerlink" href="#1-preparation-and-data-import" title="Permanent link">&para;</a></h3>
<p>Lets start by setting up the directory structure for this analysis.
The <code class="highlight">data</code> directory is where the raw sequencing reads will be. All output files will be directed to the <code class="highlight">output</code> directory. All reference files will be pointed to the <code class="highlight">reference</code> directory. The command-line scripts are stored in simple bash script format in the <code class="highlight">scripts</code> directory. For those interested, equivalent slurm scripts to run on Spartan are available in the <code class="highlight">slurm_scripts</code> directory. Although all tools are installed on the server, we will create a <code class="highlight">tools</code> directory.    </p>
<p>Lets begin by creating a byobu-screen session (see above sections for more help):
<div class="highlight"><pre><span></span><code><span class="nb">cd</span>
byobu-screen -S workshop
</code></pre></div></p>
<p>Create workshop directories:
<div class="highlight"><pre><span></span><code>mkdir data
mkdir output
mkdir reference
mkdir reference/hg38
mkdir scripts
mkdir slurm_scripts
mkdir temp
mkdir tools
</code></pre></div></p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>All analysis is being carried out in the home directory (the directory you log in to).</p>
</div>
<p>The data for this tutorial is sourced from the <a href="https://www.internationalgenome.org/data-portal/sample/NA12878">International Genome Sample Resources</a>. Raw sequencing reads from chromosome 20 are used in this tutorial. We have prepared the files which can be copied as follows:</p>
<div class="highlight"><pre><span></span><code>cp -p /mnt/shared_data/NA12878.chr20.region_1.fastq.gz data/.
cp -p /mnt/shared_data/NA12878.chr20.region_2.fastq.gz data/.
</code></pre></div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>To perform quality control checks on the raw fastq data use the tool <a href="https://www.bioinformatics.babraham.ac.uk/projects/fastqc/">FastQC</a>. Another useful QC tool output aggregator is the <a href="https://multiqc.info/">MultiQC</a> tool. MultiQC aggregates the output from several tools and outputs a single QC report for all samples. We will have a look at some of the QC data later in this section.</p>
</div>
<p>Next, we need to prepare the reference data. Luckily, we have downloaded the data and all we need to do is to create a <a href="https://kb.iu.edu/d/abbe">symbolic link</a> to the data folder as follows:</p>
<div class="highlight"><pre><span></span><code>ln -s /mnt/shared_data/* reference/hg38/.
</code></pre></div>
<p>There are several files in the reference directory. These included the GATK bundle of reference files downloaded from (<a href="ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg38/">ftp://gsapubftp-anonymous@ftp.broadinstitute.org/bundle/hg38/</a>). Additional files include in the directory are the BWA index files generated for the reference genome.</p>
<h3 id="2-align-genome">2. Align genome<a class="headerlink" href="#2-align-genome" title="Permanent link">&para;</a></h3>
<p>Run the command below to map the raw sequencing data to the Homo sapiens (human) genome assembly GRCh38 (hg38). We are using the <a href="https://github.com/lh3/bwa">BWA-MEM</a> algorithms for mapping DNA sequences against large reference genomes. Note that we have already run the created the BWA index files by running the command <code class="highlight">bwa index reference/hg38/Homo_sapiens_assembly38.fasta</code>.</p>
<p>Run BWA as follows, but first navigate to the scripts folder:</p>
<div class="highlight"><pre><span></span><code>bwa mem -M -t <span class="m">2</span> <span class="se">\</span>
-R <span class="s2">&quot;@RG\\tID:SRR622461.7\\tSM:NA12878\\tLB:ERR194147\\tPL:ILLUMINA&quot;</span> <span class="se">\</span>
reference/hg38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
data/NA12878.chr20.region_1.fastq.gz <span class="se">\</span>
data/NA12878.chr20.region_2.fastq.gz <span class="p">|</span> <span class="se">\</span>
samtools view -b -h -o output/NA12878.bam -
</code></pre></div>
<p>There are two parts to the command here. The first part uses BWA to perform the alignment and the second part take the output from BWA and uses Samtools to convert the output to the BAM format.</p>
<p>At the end of this step you should have a file called <code class="highlight">NA12878.bam</code> in the <code class="highlight">output</code> directory.</p>
<h2 id="section-2-prepare-analysis-ready-reads">Section 2: Prepare analysis ready reads<a class="headerlink" href="#section-2-prepare-analysis-ready-reads" title="Permanent link">&para;</a></h2>
<h3 id="1-sort-sambam">1. Sort SAM/BAM<a class="headerlink" href="#1-sort-sambam" title="Permanent link">&para;</a></h3>
<p>The alignment file <code class="highlight">NA12878.bam</code> is not sorted. Before proceeding, we should sort the BAM file using the <a href="https://broadinstitute.github.io/picard/">Picard</a> tools.</p>
<div class="highlight"><pre><span></span><code>picard -Xmx7g SortSam <span class="se">\</span>
<span class="nv">I</span><span class="o">=</span>output/NA12878.bam <span class="se">\</span>
<span class="nv">O</span><span class="o">=</span>output/NA12878.sort.bam <span class="se">\</span>
<span class="nv">VALIDATION_STRINGENCY</span><span class="o">=</span>LENIENT <span class="se">\</span>
<span class="nv">SORT_ORDER</span><span class="o">=</span>coordinate <span class="se">\</span>
<span class="nv">MAX_RECORDS_IN_RAM</span><span class="o">=</span><span class="m">3000000</span> <span class="se">\</span>
<span class="nv">CREATE_INDEX</span><span class="o">=</span>True    
</code></pre></div>
<p>The above command will create a coordinate sorted BAM file and an index (<code class="highlight">.bai</code>) file.</p>
<div class="admonition alignment statistics">
<p class="admonition-title">Alignment</p>
<p>Given we now have a sorted BAM file, we can now generate some useful statistics. To do so we can use the <code class="highlight">samtools flagstat</code> command. More details are available <a href="http://www.htslib.org/doc/samtools-flagstat.html">here</a>.</p>
</div>
<div class="highlight"><pre><span></span><code><span class="c1"># lets go to the home directory</span>
<span class="nb">cd</span>

samtools flagstat output/NA12878.sort.bam
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="c1"># output</span>

<span class="m">2032568</span> + <span class="m">0</span> <span class="k">in</span> total <span class="o">(</span>QC-passed reads + QC-failed reads<span class="o">)</span>
<span class="m">2030516</span> + <span class="m">0</span> primary
<span class="m">2052</span> + <span class="m">0</span> secondary
<span class="m">0</span> + <span class="m">0</span> supplementary
<span class="m">0</span> + <span class="m">0</span> duplicates
<span class="m">0</span> + <span class="m">0</span> primary duplicates
<span class="m">2032563</span> + <span class="m">0</span> mapped <span class="o">(</span><span class="m">100</span>.00% : N/A<span class="o">)</span>
<span class="m">2030511</span> + <span class="m">0</span> primary mapped <span class="o">(</span><span class="m">100</span>.00% : N/A<span class="o">)</span>
<span class="m">2030516</span> + <span class="m">0</span> paired <span class="k">in</span> sequencing
<span class="m">1015258</span> + <span class="m">0</span> read1
<span class="m">1015258</span> + <span class="m">0</span> read2
<span class="m">2030298</span> + <span class="m">0</span> properly paired <span class="o">(</span><span class="m">99</span>.99% : N/A<span class="o">)</span>
<span class="m">2030510</span> + <span class="m">0</span> with itself and mate mapped
<span class="m">1</span> + <span class="m">0</span> singletons <span class="o">(</span><span class="m">0</span>.00% : N/A<span class="o">)</span>
<span class="m">182</span> + <span class="m">0</span> with mate mapped to a different chr
<span class="m">124</span> + <span class="m">0</span> with mate mapped to a different chr <span class="o">(</span>mapQ&gt;<span class="o">=</span><span class="m">5</span><span class="o">)</span>
</code></pre></div>
<h3 id="2-mark-duplicate-reads">2. Mark duplicate reads<a class="headerlink" href="#2-mark-duplicate-reads" title="Permanent link">&para;</a></h3>
<p>The aim of this step is to locate and tag duplicate reads in the BAM file. Duplicate reads can arise due to several reasons, for more details go to <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360037052812-MarkDuplicates-Picard-">MarkDuplicates</a>.</p>
<div class="highlight"><pre><span></span><code>picard -Xmx7g MarkDuplicates <span class="se">\</span>
    <span class="nv">I</span><span class="o">=</span>output/NA12878.sort.bam <span class="se">\</span>
    <span class="nv">O</span><span class="o">=</span>output/NA12878.sort.dup.bam <span class="se">\</span>
    <span class="nv">METRICS_FILE</span><span class="o">=</span>output/marked_dup_metrics.txt
</code></pre></div>
<div class="admonition question">
<p class="admonition-title">Question: How many duplicate reads are in the duplicate marked BAM file?</p>
<details class="answer">
<summary>Answer</summary>
<p><div class="highlight"><pre><span></span><code>samtools flagstat output/NA12878.sort.dup.bam

<span class="m">2032568</span> + <span class="m">0</span> <span class="k">in</span> total <span class="o">(</span>QC-passed reads + QC-failed reads<span class="o">)</span>
<span class="m">2030516</span> + <span class="m">0</span> primary
<span class="m">2052</span> + <span class="m">0</span> secondary
<span class="m">0</span> + <span class="m">0</span> supplementary
<span class="m">7207</span> + <span class="m">0</span> duplicates
<span class="m">7207</span> + <span class="m">0</span> primary duplicates
<span class="m">2032563</span> + <span class="m">0</span> mapped <span class="o">(</span><span class="m">100</span>.00% : N/A<span class="o">)</span>
<span class="m">2030511</span> + <span class="m">0</span> primary mapped <span class="o">(</span><span class="m">100</span>.00% : N/A<span class="o">)</span>
<span class="m">2030516</span> + <span class="m">0</span> paired <span class="k">in</span> sequencing
<span class="m">1015258</span> + <span class="m">0</span> read1
<span class="m">1015258</span> + <span class="m">0</span> read2
<span class="m">2030298</span> + <span class="m">0</span> properly paired <span class="o">(</span><span class="m">99</span>.99% : N/A<span class="o">)</span>
<span class="m">2030510</span> + <span class="m">0</span> with itself and mate mapped
<span class="m">1</span> + <span class="m">0</span> singletons <span class="o">(</span><span class="m">0</span>.00% : N/A<span class="o">)</span>
<span class="m">182</span> + <span class="m">0</span> with mate mapped to a different chr
<span class="m">124</span> + <span class="m">0</span> with mate mapped to a different chr <span class="o">(</span>mapQ&gt;<span class="o">=</span><span class="m">5</span><span class="o">)</span>
</code></pre></div>
Looks like there are <strong>7207</strong> duplicate reads.</p>
</details>
</div>
<h3 id="3-base-quality-recalibration">3. Base quality recalibration<a class="headerlink" href="#3-base-quality-recalibration" title="Permanent link">&para;</a></h3>
<p>The last step of pre-processing mapped reads is the base quality score recalibration (BQSR) stage. The GATK tools detects systematic errors made by the sequencing machine while estimating the accuracy of each base. The systematic errors can be have various sources ranging from technical machine errors to the variability in the sequencing chemical reactions. The two step BQSR process applies machine learning to model the possible errors and adjust the base quality scores accordingly. More details <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035890531-Base-Quality-Score-Recalibration-BQSR-">here</a>.</p>
<div class="highlight"><pre><span></span><code><span class="c1"># lets go to the home directory again</span>
<span class="nb">cd</span>

<span class="c1"># step 1  - Build the model</span>
gatk --java-options <span class="s2">&quot;-Xmx7g&quot;</span> BaseRecalibrator <span class="se">\</span>
    -I output/NA12878.sort.dup.bam <span class="se">\</span>
    -R reference/hg38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
    --known-sites reference/hg38/dbsnp_146.hg38.vcf.gz <span class="se">\</span>
    -O output/recal_data.table
</code></pre></div>
<div class="highlight"><pre><span></span><code><span class="c1"># step 2: Apply the model to adjust the base quality scores</span>
gatk --java-options <span class="s2">&quot;-Xmx7g&quot;</span> ApplyBQSR <span class="se">\</span>
    -I output/NA12878.sort.dup.bam <span class="se">\</span>
    -R reference/hg38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
    --bqsr-recal-file output/recal_data.table <span class="se">\</span>
    -O output/NA12878.sort.dup.bqsr.bam
</code></pre></div>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>In a workflow such as this it is a good practice to give the output files an appropriate name. In this case, we are appending the workflow step details to the filenames. For example, append <code>dup</code> after running the mark duplicates step.</p>
</div>
<p>We now have a pre-processed BAM file (<code class="highlight">NA12878.sort.dup.bqsr.bam</code>) ready for variant calling.</p>
<p>But before we proceed, let&rsquo;s take a detour and run some summary statistics of the alignment data and QC.</p>
<details class="example">
<summary><strong>BAM statistics and QC</strong> </summary>
<p>The commands below uses FastQC and Picard to generate QC metrics followed by multiQC tools then aggregating the data to produce an HTML report.    </p>
<p><div class="highlight"><pre><span></span><code><span class="c1"># FastQC</span>
fastqc data/NA12878.chr20.region_1.fastq.gz <span class="se">\</span>
data/NA12878.chr20.region_2.fastq.gz <span class="se">\</span>
-o output/

<span class="c1"># CollectInsertSizeMetrics</span>
picard CollectMultipleMetrics <span class="se">\</span>
<span class="nv">R</span><span class="o">=</span>reference/hg38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
<span class="nv">I</span><span class="o">=</span>output/NA12878.sort.dup.bqsr.bam <span class="se">\</span>
<span class="nv">O</span><span class="o">=</span>output/NA12878.sort.dup.bqsr.CollectMultipleMetrics

<span class="c1"># MultiQC</span>
multiqc output/. -o output/.    
</code></pre></div>
We have precomputed this and the resulting MultiQC report is <a href="../files/multiqc_report.html" target="_blank">here</a>.</p>
</details>
<h2 id="section-3-variant-calling">Section 3: Variant calling<a class="headerlink" href="#section-3-variant-calling" title="Permanent link">&para;</a></h2>
<p>The next step in the GATK best practices workflow is to proceed with the variant calling.</p>
<p>There are a couple of workflows to call variants using GATK4. Here we will follow the Genomic Variant Call Format (GVCF) workflow which is more suited for scalable variant calling i.e. allows incremental addition of samples for joint genotyping.</p>
<h3 id="1-apply-haplotypecaller">1. Apply HaplotypeCaller<a class="headerlink" href="#1-apply-haplotypecaller" title="Permanent link">&para;</a></h3>
<p>HaplotypeCaller is the focal tool within GATK4 to simultaneously call germline SNVs and small Indels using local de-novo assembly of haplotype regions.</p>
<div class="admonition algorithm">
<p class="admonition-title">Algorithm</p>
<p>Briefly, the HaplotypeCaller works by:
1. Identify active regions or regions with evidence of variations.
2. Re-asssemble the active regions.
3. Re-align active region reads to the assembled regions to identify allele.
More details about the HaplotypeCaller can be found <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360037225632-HaplotypeCaller">here</a>.</p>
</div>
<div class="highlight"><pre><span></span><code>gatk --java-options <span class="s2">&quot;-Xmx7g&quot;</span> HaplotypeCaller <span class="se">\</span>
    -I output/NA12878.sort.dup.bqsr.bam <span class="se">\</span>
    -R reference/hg38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
    -ERC GVCF <span class="se">\</span>
    -L chr20 <span class="se">\</span>
    -O output/NA12878.g.vcf.gz
</code></pre></div>
<p>The output of this step is a GVCF file. The format for the GVCF file is similar to a VCF file. The key difference is that the GVCF file contains records for each sequenced genomic coordinate. The <code class="highlight">--emit-ref-confidence</code> or <code class="highlight">-ERC</code> parameter lets you select a method to summarise confidence in the genomic site being homozygous-reference. The option <code class="highlight">-ERC GVCF</code> is more efficient and recommended for large samples and therefore more scalable.</p>
<h3 id="2-apply-combinegvcfs">2. Apply CombineGVCFs<a class="headerlink" href="#2-apply-combinegvcfs" title="Permanent link">&para;</a></h3>
<p>The CombineGVCFs tool is applied to combine multiple single sample GVCF files to merge these in to a single multi-sample GVCF file.</p>
<p>We have pre-processed two additional samples (NA12891 and NA12892) up to the HaplotypeCaller step (above). Lets first copy the gvcf files to the output directory.</p>
<div class="highlight"><pre><span></span><code><span class="c1">#lets make sure that we are in the apropriate directory</span>
<span class="nb">cd</span>

cp /mnt/shared_data/NA12891.g.vcf.gz* output/.
cp /mnt/shared_data/NA12892.g.vcf.gz* output/.
</code></pre></div>
<div class="highlight"><pre><span></span><code>gatk --java-options <span class="s2">&quot;-Xmx7g&quot;</span> CombineGVCFs <span class="se">\</span>
    -R reference/hg38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
    -V output/NA12878.g.vcf.gz <span class="se">\</span>
    -V output/NA12891.g.vcf.gz <span class="se">\</span>
    -V output/NA12892.g.vcf.gz <span class="se">\</span>
    -L chr20 <span class="se">\</span>
    -O output/cohort.g.vcf.gz
</code></pre></div>
<details class="example">
<summary>Lets look at the combined gvcf file</summary>
<p><div class="highlight"><pre><span></span><code>less output/cohort.g.vcf.gz
</code></pre></div>
Work your way down to the variant records? How many samples do you see in the VCF file?
Hint: look at the header row.</p>
</details>
<p>Now that we have a merged GVCF file, we are ready to perform genotyping.</p>
<h3 id="3-apply-genotypegvcfs">3. Apply GenotypeGVCFs<a class="headerlink" href="#3-apply-genotypegvcfs" title="Permanent link">&para;</a></h3>
<p>GenotypeGVCFs</p>
<div class="highlight"><pre><span></span><code>gatk --java-options <span class="s2">&quot;-Xmx7g&quot;</span> GenotypeGVCFs <span class="se">\</span>
    -R reference/hg38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
    -V output/cohort.g.vcf.gz <span class="se">\</span>
    -L chr20 <span class="se">\</span>
    -O output/output.vcf.gz
</code></pre></div>
<details class="information">
<summary>Information</summary>
<p>An alternative to CombineGVCFs is <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360057439331-GenomicsDBImport">GenomicsDBImport</a>, which is more efficient for sample numbers and stores the content in a a GenomicsDB data store. Therefore, CombineGVCFs could be slow and inefficient for more than a few samples. A possible work around is to split up the tasks per interval regions such as chromosomes.</p>
</details>
<details class="example">
<summary>Visualisations: VCF file</summary>
<p>Screenshot from output.vcf.gz
<img alt="fig1" src="../media/fig1.png" /></p>
</details>
<h2 id="section-4-filter-and-prepare-analysis-ready-variants">Section 4: Filter and prepare analysis ready variants<a class="headerlink" href="#section-4-filter-and-prepare-analysis-ready-variants" title="Permanent link">&para;</a></h2>
<h3 id="1-variant-quality-score-recalibration">1. Variant Quality Score Recalibration<a class="headerlink" href="#1-variant-quality-score-recalibration" title="Permanent link">&para;</a></h3>
<p>The raw VCF file from the previous step (<code class="highlight">output.vcf.gz</code>) contains 10467 variants. Not all of these are real, therefore, the aim of this step is to filter out artifacts or false positive variants. GATK has provided different workflows for variant filtering. Here we will walk through the Variant Quality Score Recalibration or the VQSR strategy. VQSR is a two step process (1) the first step builds a model that describes how variant metric or quality measures co-vary with the known variants in the training set. (2) The second step then ranks each variant according to the target sensitivity cutoff and applies a filter expression.</p>
<div class="highlight"><pre><span></span><code><span class="c1">#Step 1 - VariantRecalibrator</span>
gatk --java-options <span class="s2">&quot;-Xmx7g&quot;</span> VariantRecalibrator <span class="se">\</span>
    -V output/output.vcf.gz <span class="se">\</span>
    --trust-all-polymorphic <span class="se">\</span>
    -mode SNP <span class="se">\</span>
    --max-gaussians <span class="m">6</span> <span class="se">\</span>
    --resource:hapmap,known<span class="o">=</span>false,training<span class="o">=</span>true,truth<span class="o">=</span>true,prior<span class="o">=</span><span class="m">15</span> reference/hg38/hapmap_3.3.hg38.vcf.gz <span class="se">\</span>
    --resource:omni,known<span class="o">=</span>false,training<span class="o">=</span>true,truth<span class="o">=</span>true,prior<span class="o">=</span><span class="m">12</span> reference/hg38/1000G_omni2.5.hg38.vcf.gz <span class="se">\</span>
    --resource:1000G,known<span class="o">=</span>false,training<span class="o">=</span>true,truth<span class="o">=</span>false,prior<span class="o">=</span><span class="m">10</span> reference/hg38/1000G_phase1.snps.high_confidence.hg38.vcf.gz <span class="se">\</span>
    --resource:dbsnp,known<span class="o">=</span>true,training<span class="o">=</span>false,truth<span class="o">=</span>false,prior<span class="o">=</span><span class="m">7</span> reference/hg38/dbsnp_138.hg38.vcf.gz <span class="se">\</span>
    -an QD -an MQRankSum -an ReadPosRankSum -an FS -an MQ -an SOR -an DP <span class="se">\</span>
    -O output/cohort_snps.recal <span class="se">\</span>
    --tranches-file output/cohort_snps.tranches

<span class="c1">#Step 2 - ApplyVQSR</span>
gatk --java-options <span class="s2">&quot;-Xmx7g&quot;</span> ApplyVQSR <span class="se">\</span>
    -R reference/hg38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
    -V output/output.vcf.gz <span class="se">\</span>
    -O output/output.vqsr.vcf <span class="se">\</span>
    --truth-sensitivity-filter-level <span class="m">99</span>.0 <span class="se">\</span>
    --tranches-file output/cohort_snps.tranches <span class="se">\</span>
    --recal-file output/cohort_snps.recal <span class="se">\</span>
    -mode SNP
</code></pre></div>
<div class="admonition countvariants">
<p class="admonition-title">Countvariants</p>
<p>There are number of ways to count the variants in a VCF file. A very straight forward way using the GATK4 tools is as follows:
<div class="highlight"><pre><span></span><code>gatk CountVariants -V output/output.vqsr.vcf
</code></pre></div></p>
<div class="highlight"><pre><span></span><code>Tool returned:
10467
</code></pre></div>
</div>
<p>There are several protocols for filtering VCF files. We have walked throught the VQSR strategy above and for other options please visit this <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360035531112--How-to-Filter-variants-either-with-VQSR-or-by-hard-filtering">link</a>.</p>
<details class="example">
<summary>Filtering strategy for a single sample VCF file</summary>
<p>Consider the following method to filter a single sample VCF file. Here we will go through the Convolutional Neural Net based protocol to annotate and filter the VCF file.</p>
<p>This is a two step process:</p>
<p>(i) CNNScoreVariants will annotate the variant with pre-computed single-sample derived model scores in the INFO field CNN_1D (the neural network performs convolutions over the reference sequence surrounding the variant and combines those features with a multilayer perceptron on the variant annotations).</p>
<div class="highlight"><pre><span></span><code>gatk --java-options <span class="s2">&quot;-Xmx7g&quot;</span> CNNScoreVariants  <span class="se">\</span>
   -R reference/hg38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
   -V output/output.vcf.gz <span class="se">\</span>
   -O output/output.cnns.vcf
</code></pre></div>
<p>(ii) FilterVariantTranches takes as input the percent sensitivities (0-100) to known sites to apply the filter. Variants with scores higher than for e.g. 99<sup>th</sup> percentile of variants in the resources pass through the filter and will have <code>PASS</code> in the filter. Others will have a filter values like &lsquo;CNN_1D_INDEL_Tranche_99.40_100.00&rsquo; or &lsquo;CNN_1D_SNP_Tranche_99.95_100.00&rsquo;.</p>
<div class="highlight"><pre><span></span><code>gatk --java-options <span class="s2">&quot;-Xmx7g&quot;</span> FilterVariantTranches <span class="se">\</span>
    -V output/output.cnns.vcf <span class="se">\</span>
    --resource reference/hg38/hapmap_3.3.hg38.vcf.gz <span class="se">\</span>
    --resource reference/hg38/Mills_and_1000G_gold_standard.indels.hg38.vcf.gz <span class="se">\</span>
    --info-key CNN_1D <span class="se">\</span>
    --snp-tranche <span class="m">99</span>.95 <span class="se">\</span>
    --indel-tranche <span class="m">99</span>.4 <span class="se">\</span>
    -O output/output.cnns.cnnfilter.vcf
</code></pre></div>
</details>
<div class="admonition hint">
<p class="admonition-title">Hint</p>
<p>BCFtools is a useful tool to manipulate, filter and query VCF files. More details from <a href="https://samtools.github.io/bcftools/">BCFtools</a>. BCFtools can be combined with linux command line tools as well to summarise data. For example, the command below can used extract and print the &lsquo;FILTER&rsquo; column from the VCF file.</p>
<div class="highlight"><pre><span></span><code>    bcftools query -f<span class="s1">&#39;%FILTER\n&#39;</span> output/output.vqsr.vcf.gz
</code></pre></div>
</div>
<h3 id="2-additional-filtering">2. Additional filtering<a class="headerlink" href="#2-additional-filtering" title="Permanent link">&para;</a></h3>
<p>The VariantFiltration tools is designed for hard-filtering variant calls based on custom quality criteria such as sequencing depth, mapping quality etc. The two parameters are the filter-name and filter-expression. The parameter filter-name is the name of the filter to be used in the FILTER column if the expression in filter-expression is true. In the example below, if the sequencing depth at the variant site (VCF field DP) is less than 10, the FILTER field will be populated with the value &lsquo;Low_depth10&rsquo;. Users can add multiple filter expression/name combinations.</p>
<div class="highlight"><pre><span></span><code>gatk --java-options <span class="s2">&quot;-Xmx7g&quot;</span> VariantFiltration <span class="se">\</span>
    -R reference/hg38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
    -V output/output.vqsr.vcf <span class="se">\</span>
    -O output/output.vqsr.varfilter.vcf <span class="se">\</span>
    --filter-name <span class="s2">&quot;Low_depth10&quot;</span> <span class="se">\</span>
    --filter-expression <span class="s2">&quot;DP &lt; 10&quot;</span>
</code></pre></div>
<div class="admonition question">
<p class="admonition-title">Question: How many variants have a low sequencing depth (DP&lt;10) in the file output.vqsr.varfilter.vcf.</p>
<details class="answer">
<summary>Answer</summary>
<div class="highlight"><pre><span></span><code>    bcftools query -f<span class="s1">&#39;%FILTER\n&#39;</span> output/output.vqsr.varfilter.vcf <span class="p">|</span> sort <span class="p">|</span> uniq -c
</code></pre></div>
<div class="highlight"><pre><span></span><code>  6 Low_depth10
  2 Low_depth10;VQSRTrancheSNP99.00to99.90
  9 Low_depth10;VQSRTrancheSNP99.90to100.00
  9064 PASS
  1278 VQSRTrancheSNP99.00to99.90
  108 VQSRTrancheSNP99.90to100.00
</code></pre></div>
</details>
</div>
<h3 id="3-final-analysis-ready-vcf-file">3. Final analysis ready VCF file<a class="headerlink" href="#3-final-analysis-ready-vcf-file" title="Permanent link">&para;</a></h3>
<p>Given we have a filter annotated VCF files (), we can now create an analysis ready VCF file.</p>
<div class="admonition question">
<p class="admonition-title">Question: Create a VCF file called <code>output/output.vqsr.varfilter.pass.vcf.gz</code> that contains only PASS variants? The input VCF file is <code>output/output.vqsr.varfilter.vcf</code>.&rdquo; Hint: try using the Bcftools application.</p>
<details class="answer">
<summary>Answer</summary>
<p>Use the bcftools to filter PASS variants.
<div class="highlight"><pre><span></span><code>bcftools view -f <span class="s1">&#39;PASS,.&#39;</span> -O vcf -o output/output.vqsr.varfilter.pass.vcf output/output.vqsr.varfilter.vcf            
</code></pre></div></p>
<p>We have now created an analysis ready version of the VCF file. It is also a good practice to compress and index the file.</p>
<div class="highlight"><pre><span></span><code>bgzip -c output/output.vqsr.varfilter.pass.vcf &gt; output/output.vqsr.varfilter.pass.vcf.gz
tabix -p vcf output/output.vqsr.varfilter.pass.vcf.gz
</code></pre></div>
</details>
</div>
<h2 id="section-5-exporting-variant-data-and-visualisation">Section 5: Exporting variant data and visualisation<a class="headerlink" href="#section-5-exporting-variant-data-and-visualisation" title="Permanent link">&para;</a></h2>
<p>VCF files, although in tabular format, are not user friendly. We will go through a couple of ways to share share and visualise variant data. This is important for downstream analysis as well as sharing data. First, we will convert the VCF file in to a TSV file (ready for Excel for example) in a manner where we extract data fields of interest.</p>
<h3 id="1-variantstotable">1. VariantsToTable<a class="headerlink" href="#1-variantstotable" title="Permanent link">&para;</a></h3>
<p>This GATK4 tool extracts fields of interest from each record in a VCF file. <a href="https://gatk.broadinstitute.org/hc/en-us/articles/360056968292-VariantsToTable">VariantsToTable</a> can extract field from both the INFO and FORMAT columns in the VCF file.</p>
<div class="admonition note">
<p class="admonition-title">Note</p>
<p>VariantsToTable, by default, only extracts PASS or . (no filtering applied) variants. Use the <code class="highlight">--show-filtered</code> parameter to show all variants.</p>
</div>
<div class="highlight"><pre><span></span><code>gatk VariantsToTable <span class="se">\</span>
    -R reference/hg38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
    -V output/output.vqsr.varfilter.pass.vcf.gz <span class="se">\</span>
    -F CHROM -F POS -F FILTER -F TYPE -GF AD -GF DP <span class="se">\</span>
    --show-filtered <span class="se">\</span>
    -O output/output.vqsr.varfilter.pass.tsv
</code></pre></div>
<h3 id="2-html-report">2. HTML report<a class="headerlink" href="#2-html-report" title="Permanent link">&para;</a></h3>
<p>Another useful method for sharing data is an interactive HTML file. This is suited for sharing a smaller subset of variants along with sequencing data. Here we will go through a simple example using the <a href="https://github.com/brentp/jigv">jigv</a> tool.</p>
<p><img alt="fig2" src="../media/fig2.png" /></p>
<p>We will start with creating a subset of variants to report.</p>
<div class="highlight"><pre><span></span><code>bcftools view output/output.vqsr.varfilter.pass.vcf.gz <span class="se">\</span>
chr20:3822018-3999324 <span class="p">|</span> <span class="se">\</span>
bgzip -c &gt; output/subset.vcf.gz

tabix -p vcf output/subset.vcf.gz
</code></pre></div>
<p>Now, we will call the jigv tool command to generate the report.
<div class="highlight"><pre><span></span><code>jigv --sample NA12878 <span class="se">\</span>
--sites output/subset.vcf.gz <span class="se">\</span>
--fasta reference/hg38/Homo_sapiens_assembly38.fasta <span class="se">\</span>
output/NA12878.sort.dup.bqsr.bam &gt; output/NA12878.jigv.html
</code></pre></div></p>
<p>Here is an example <a href="../files/NA12878.html" target="_blank">report</a> we created earlier.</p>
<!-- ```bash
jigv --sample NA12878 --ped family.ped --sites output/subset.vcf.gz --fasta reference/hg38/Homo_sapiens_assembly38.fasta output/*.sort.dup.bqsr.bam > output/NA12878.jigv.html
``` -->

              
            </article>
          </div>
        </div>
        
      </main>
      
        <footer class="md-footer">
  
  <div class="md-footer-meta md-typeset">
    <div class="md-footer-meta__inner md-grid">
      <div class="md-copyright">
  
    <div class="md-copyright__highlight">
      <a href="https://www.melbournebioinformatics.org.au/">Melbourne Bioinformatics</a>
    </div>
  
  
    Made with
    <a href="https://squidfunk.github.io/mkdocs-material/" target="_blank" rel="noopener">
      Material for MkDocs
    </a>
  
</div>
      
    </div>
  </div>
</footer>
      
    </div>
    <div class="md-dialog" data-md-component="dialog">
      <div class="md-dialog__inner md-typeset"></div>
    </div>
    <script id="__config" type="application/json">{"base": "../../..", "features": [], "search": "../../../assets/javascripts/workers/search.2a1c317c.min.js", "translations": {"clipboard.copied": "Copied to clipboard", "clipboard.copy": "Copy to clipboard", "search.config.lang": "en", "search.config.pipeline": "trimmer, stopWordFilter", "search.config.separator": "[\\s\\-]+", "search.placeholder": "Search", "search.result.more.one": "1 more on this page", "search.result.more.other": "# more on this page", "search.result.none": "No matching documents", "search.result.one": "1 matching document", "search.result.other": "# matching documents", "search.result.placeholder": "Type to start searching", "search.result.term.missing": "Missing", "select.version.title": "Select version"}}</script>
    
    
      <script src="../../../assets/javascripts/bundle.748e2769.min.js"></script>
      
    
  </body>
</html>